tests/thread_tests/thread_test.go

```diff
@@ -18,17 +17,15 @@ func TestAllClustersAppThread(t *testing.T) {
 
 	t.Run("Commission", func(t *testing.T) {
 		stdout, _, _ := utils.Exec(t, "chip-tool pairing code-thread 110 hex:"+trimmedActiveDataset+" 34970112332 2>&1")
-		assert.NoError(t,
-			os.WriteFile("chip-tool-thread-pairing.log", []byte(stdout), 0644),
-		)
+
+		assert.NoError(t, utils.WriteLogFile(t, "chip-tool-thread-pairing", stdout))
```

Comment:
Given that `utils.WriteLogFile` uses the test's name in the file, aren't the labels mostly redundant?

Reply:
The other tests use the same approach to log file naming. I agree that this is unnecessary duplication which we can clean up. Maybe rather in a separate PR?
```
$ ls logs
TestAllClustersAppWiFi-Commission-chip-tool-pairing.log
TestAllClustersAppWiFi-Control-chip-tool-toggle.log
TestAllClustersAppWiFi-matter-all-clusters-app.log

TestUpgrade-Commission-chip-tool-pairing.log
TestUpgrade-Control_before_upgrade-chip-tool-onoff.log
TestUpgrade-Control_upgraded_snap-chip-tool-onoff.log
TestUpgrade-matter-all-clusters-app.log

$ ls thread_tests/logs
TestAllClustersAppThread-Commission-chip-tool-thread-pairing.log
TestAllClustersAppThread-Control-chip-tool-thread-onoff.log
```

Reply:
I've removed the custom string labels and replaced them with the snap names.

```
$ ls logs
TestAllClustersAppWiFi-Commission-chip-tool.log
TestAllClustersAppWiFi-Control-chip-tool.log
TestAllClustersAppWiFi-matter-all-clusters-app.log

TestUpgrade-Commission-chip-tool.log
TestUpgrade-Control_before_upgrade-chip-tool.log
TestUpgrade-Control_after_upgrade-chip-tool.log
TestUpgrade-matter-all-clusters-app.log

$ ls thread_tests/logs/
TestAllClustersAppThread-Commission-chip-tool.log
TestAllClustersAppThread-Control-chip-tool.log
```